test-integration/helpers/student-dashboard.coffee

```diff
@@ -0,0 +1,100 @@
+_ = require 'underscore'
+CourseSelect = require './course-select'
+
+# A base class for a "Panel" type DOM object
+# If anyone finds this useful, feel free to expand and extract to helpers
+class PanelElement
+
+  constructor: (test, el) ->
+    @_test = test
+    if _.isString(el)
+      el = @test.driver.findElement(css: el)
```

Comment:
so would the logic in the test need to be something like:

```
if !dash.valid()  # eep, could this be prefixed with `is` so it is not confusing whether it is a getter or performing an action
  dash = new StudentDashboard(@, 'student01', ...)
```

if the test logged in as a student, logged out and back as a teacher, and then logged out and back as a student?